File: Stepik/Функциональное программирование на языке Haskell/3.1 - Функции для работы со списками/task5.hs
sum3 :: Num a => [a] -> [a] -> [a] -> [a]

sum3 [] [] [] = []

sum3 (x:xs) [] [] = x : sum3 xs [] []
sum3 [] (y:ys) [] = y : sum3 [] ys []
sum3 [] [] (z:zs) = z : sum3 [] [] zs

sum3 (x:xs) (y:ys) []     = (x + y) : sum3 xs ys []
sum3 (x:xs) []     (z:zs) = (x + z) : sum3 xs [] zs
sum3 []     (y:ys) (z:zs) = (y + z) : sum3 [] ys zs

sum3 (x:xs) (y:ys) (z:zs) = (x + y + z) : sum3 xs ys zs

File: Stepik/Программирование на Golang/1.12 - Массивы и срезы/task5/task5.go
package main

import "fmt"

func main() {
	var N int
	fmt.Scan(&N)

	var slice []int
	for i := 0; i < N; i++ {
		var num int
		fmt.Scan(&num)
		slice = append(slice, num)
	}

	var count int = 0
	for i := 0; i < len(slice); i++ {
		if slice[i] > 0 {
			count++
		}
	}

	fmt.Println(count)
}
